### DISCUSSION TITLE
Erro na montagem do formulário bricks

### DISCUSSION THREAD

**Discussion by FranciscoMaik** [mercadopago/sdk-react#62] [General]
Olá, tivemos um problema com a montagem do componente de formulário bricks, gerando duplicidade no formulário, mas observamos que isso acontece somente quando a internet é um pouco lenta, devido as validações dos campos de segurança.

Temos como retorno o seguinte error: 
```javascript
{"type":"critical","cause":"fields_setup_failed_after_3_tries","message":"The integration with Secure Fields `failed"}
``` 

Imagem de exemplo: 
![image](https://github.com/mercadopago/sdk-react/assets/20601076/0e11b33b-6d03-40ea-9dc9-b29cd2349801)

Suspeitamos que possa ser no momento das validações, procuramos nas documentação mas não encontramos se é possível esperar essa primeira validação antes de realizar outra, já que pelo erro percebemos que está sendo realizado mais de uma vez. Seria possível validar isso.

**Comment by ghost** [comment]
Olá @FranciscoMaik, estamos analisando o erro reportado, assim que tiver novidades damos um retorno aqui.

**Comment by ghost** [comment]
@FranciscoMaik percebemos um bug no Card Payment Brick onde a tela de erro (à direita no print que você enviou) não sobrepõe o próprio componente do brick, estamos trabalhando na correção e pretendemos colocar em produção na semana que vem.
Sobre o erro `fields_setup_failed_after_3_tries`, ele não é causado por validações, mas sim pela montagem dos campos de cartão (número, data de vencimento e código de segurança). Este erro está acontecendo com qual frequência com vocês? Mesmo com a internet lenta, deveria funcionar pois temos cache para esses campos.

**Reply by FranciscoMaik** [reply]
Olá @andreagostinho-meli, sobre a frequência estava sendo sempre que montava o componente com a internet um pouco lenta, mas identificamos que o problema se deu por causa de um dado da variável de _initialization_ que sofria uma atualização depois da montagem do componente, peço desculpas pelo inconveniente. Espero que esse tópico possa ajudar alguém a não cometer o mesmo erro que eu. Muito obrigado pelo retorno;

**Reply by ghost** [reply]
@FranciscoMaik entendi, qual dado seria? temos interesse de saber para nos ajudar em questões de troubleshooting.

**Reply by FranciscoMaik** [reply]
O valor de _amount_.

**Reply by ghost** [reply]
@FranciscoMaik certo, como você estava fazendo a atualização do `amount`? Atualmente não é possível atualizá-lo após a montagem do componente, imagino que estava renderizando o componente novamente sem chamar o método `unmount`.
Chamando este método deve funcionar sua atualização do `amount` porque o componente será destruído antes de ser renderizado com o novo valor de `amount`.

```js
window.cardPaymentBrickController.unmount()
```
- [Documentação no site de desenvolvedores](https://www.mercadopago.com.br/developers/pt/docs/checkout-bricks/card-payment-brick/default-rendering)
- [Documentação no Github](https://github.com/mercadopago/sdk-js/blob/main/API/bricks/card-payment.md#brick-controllerunmount)
